### PR TITLE
README: Cleanup, add Versioning and Update sections

### PR DESCRIPTION
**Pull request by akien-mga**
This clarifies the correspondance between this repo's branches and the upstream
ones, as well as documents the update workflow used by maintainers to sync the
headers, which can be used by anyone for their own builds (or if they're
running cutting edge Godot snapshots before this repo gets updated).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,46 @@
 # godot-headers
 
-#### `GDNative / NativeScript`
+This repository contains C headers for [**Godot Engine**](
+https://github.com/godotengine/godot)'s *GDNative* API, which
+can be used to write *NativeScripts*.
 
-> `GDNative` enables the use of dynamically linked libraries inside of [**Godot**](https://github.com/godotengine/godot).
+> `GDNative` enables the use of dynamically linked libraries inside o
+[**Godot**](https://github.com/godotengine/godot).
```
Comment (Calinou) [inline README.md:8]:
```suggestion
> `GDNative` enables the use of dynamically linked libraries inside of
> [**Godot**](https://github.com/godotengine/godot).
```

File: README.md [hunk 2]
```diff
@@ -1,20 +1,46 @@
 # godot-headers
 
-#### `GDNative / NativeScript`
+This repository contains C headers for [**Godot Engine**](
+https://github.com/godotengine/godot)'s *GDNative* API, which
+can be used to write *NativeScripts*.
```
Comment (Calinou) [inline README.md:5]:
```suggestion
This repository contains C headers for
[**Godot Engine**](https://github.com/godotengine/godot)'s *GDNative* API,
which can be used to write *NativeScripts*.
```

File: README.md [hunk 3]
```diff
@@ -1,20 +1,46 @@
...
+  the headers in this repository may not always be fully in sync with upstream.
+  See [**Updating Headers**](#updating-headers) if you need to bring
+  them up to date.
+- `3.x` tracks the development of the next 3.x minor release. Like `master`, it
+  might not always be fully up-to-date with upstream.
+- Other versioned branches (e.g. `3.3`, `3.2`) track the latest stable release
+  in the corresponding branch.
+
+Stable releases are also tagged on this repository: [**Tags**](
+https://github.com/godotengine/godot-headers/tags).
```
Comment (Calinou) [inline README.md:32]:
```suggestion
Stable releases are also tagged on this repository:
[**Tags**](https://github.com/godotengine/godot-headers/tags).
```

File: README.md [hunk 4]
```diff
@@ -74,10 +101,10 @@ void GDN_EXPORT godot_gdnative_terminate(godot_gdnative_terminate_options *o) {
 
 /** Script entry (Registering all the classes and stuff) **/
```
Comment (Calinou) [inline README.md:102]: Are we sure we want to use 8-space indentation here?

**Review by Calinou — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (akien-mga) [reply README.md:102]: I just followed the existing usage to minimize the diff. But yes I can switch it to tabs or 4-spaces if that's better.

Comment (akien-mga) [reply README.md]: I went with tabs since most of it is C code, and even for GDScript the editor defaults to tabs still.

**Review by BastiaanOlij — APPROVED** [review]
Looking good to me! lets go!

**Pull request merged**